framework/src/timesteppers/Stepper.C

```diff
@@ -0,0 +1,748 @@
+  return new IfBlock(nth, between, [=](StepperInfo & si)
+                     {
+                       return (si.stepCount() + (every_n - offset) - 1) % every_n == 0;
+                     });
+}
+
+StepperBlock *
+BaseStepper::initialN(StepperBlock * initial, StepperBlock * primary, int n)
+{
+  return new IfBlock(initial, primary, [=](StepperInfo & si)
+                     {
+                       return si.stepCount() <= n;
+                     });
+}
+
+StepperBlock *
+BaseStepper::converged(StepperBlock * converged, StepperBlock * not_converged, bool delay)
+{
+  return new IfBlock(converged, not_converged, [=](StepperInfo & si)
+                     {
+                       return si.converged() && (!delay || si.converged(1));
+                     });
+}
+
+StepperBlock *
+BaseStepper::min(StepperBlock * a, StepperBlock * b, Real tol)
+{
+  return new MinOfBlock(a, b, tol);
+}
+
+RootBlock::RootBlock(std::function<Real(StepperInfo & si)> func)
+  : _func(func)
+{
+}
+
+Real
+RootBlock::next(StepperInfo & si)
+{
+  return _func(si);
+}
+
+ModBlock::ModBlock(StepperBlock * s,
+                   std::function<
+                       Real(StepperInfo & si, Real dt)> func)
+  : _stepper(s), _func(func)
+{
+}
+
+Real
+ModBlock::next(StepperInfo & si)
+{
+  return _func(si, _stepper->next(si));
+}
+
+IfBlock::IfBlock(StepperBlock * on_true, StepperBlock * on_false,
+                 std::function<bool(StepperInfo &)> func)
+  : _ontrue(on_true), _onfalse(on_false), _func(func)
+{
+}
+
+Real
+IfBlock::next(StepperInfo & si)
+{
+  bool val = _func(si);
+  if (val)
+    return _ontrue->next(si);
+  return _onfalse->next(si);
+}
+
+
+InstrumentedBlock::InstrumentedBlock(Real * dt_store)
+  : _stepper(nullptr), _dt_store(dt_store), _own(!dt_store)
+{
+  if (!_dt_store)
+    _dt_store = new Real(0);
+}
+
+InstrumentedBlock::~InstrumentedBlock()
+{
+  if (_own)
+    delete _dt_store;
+}
+
+Real
+InstrumentedBlock::next(StepperInfo & si)
+{
+  if (!_stepper)
+    mooseError("InstrumentedStepper's inner stepper not set");
+
+  *_dt_store = _stepper->next(si);
+  return *_dt_store;
+}
+
+void
+InstrumentedBlock::setStepper(StepperBlock * s)
+{
+  _stepper.reset(s);
+}
+
+Real *
+InstrumentedBlock::dtPtr()
+{
+  return _dt_store;
+}
+
+RetryUnusedBlock::RetryUnusedBlock(StepperBlock * s, Real tol, bool prev_prev)
+  : _stepper(s), _tol(tol), _prev_prev(prev_prev), _prev_dt(0), _prev_time(0)
+{
+}
+
+Real
+RetryUnusedBlock::next(StepperInfo & si)
+{
+  if (_prev_dt != 0 && std::abs(si.dt() - _prev_dt) > _tol && std::abs(si.time() - _prev_time) > _tol)
+  {
+    if (_prev_prev)
+    {
+      _prev_dt = si.dt(1);
+      return si.dt(1);
+    }
+    else
+      return _prev_dt;
+  }
+  _prev_time = si.time();
+  _prev_dt = _stepper->next(si);
+  return _prev_dt;
+}
+
+ConstrFuncBlock::ConstrFuncBlock(StepperBlock * s, std::function<Real(Real)> func,
+                                 Real max_diff)
+  : _stepper(s), _func(func), _max_diff(max_diff)
```

Comment:
This doesn't follow our formatting convention. Can clang-format deal with this? I.e. we have arguments to methods stacked on lines but initialization lists are _always_ one per line.